src/workerd/util/thread-scopes.h

```diff
@@ -99,4 +99,14 @@ class ThreadProgressCounter {
   friend class Watchdog;
 };
 
+class NoProxyLoggingScope {
```

Comment:
Needs comments.

Maybe the name should indicate that this is very specific to the CF blob.